### PR TITLE
Metadata error

### PR DESCRIPTION
**Pull request by markphillippebworth**
Fixing issue where it fails if there are NAs in the cell type column and added a check to see whether the cell types in question were  present or not within the data.

### CODE REVIEW

File: R/callOpenTiles.R [hunk 1]
```diff
@@ -375,9 +375,16 @@ setMethod(
...
-  } else {
+  } else if(any(rownames(allCellCounts) %in% cellPopulations)){
     allCellCounts <- allCellCounts[rownames(allCellCounts) %in% cellPopulations, , drop = FALSE]
     allFragmentCounts <- allFragmentCounts[rownames(allFragmentCounts) %in% cellPopulations, , drop = FALSE]
+  } else{
+  
+    stop(
+      stringr::str_interp("Some or all of the cell populations provided were not found in the metadata column ${cellPopLabel}. 
+        These cell populations include ${cellPopulations}.")
+    )
```
Comment (imran-aifi) [inline R/callOpenTiles.R:386]: This kind of check should happen earlier, like in to the top of `.callOpenTiles_default`.

Comment (imran-aifi) [reply R/callOpenTiles.R:386]: It should: check that all of `cellPopulations` exists in `unique(cellColData[[cellPopLabel]])`

File: R/callOpenTiles.R [hunk 2]
```diff
@@ -390,6 +397,14 @@ setMethod(
...
   # !!as.name(cellPopLabel) %in% cellPopulations))
   cellColDataCopy <- copy(cellColData)
+    
+  if(any(is.na(cellColDataCopy[[cellPopLabel]]))){
+  
+      warning(
+          stringr::str_interp("Some cells within the column ${cellPopLabel} are labeled as NA. Those cells will be ignored.")
+        )
+      
+  }
```
Comment (imran-aifi) [inline R/callOpenTiles.R:407]: This warning can also be moved earlier, and check `cellColData` (not the copy). 
can be moved to the top of `.callOpenTiles_default`.

File: R/callOpenTiles.R [hunk 3]
```diff
@@ -420,6 +435,11 @@ setMethod(
       )
 
       summarizedData <- as.data.frame(summarizedData)
+        
+      #Remove any columns with missing labels
+      cellTypeList <- summarizedData[[cellPopLabel]]
+      summarizedData <- summarizedData[!is.na(cellTypeList),,drop=FALSE]
+      cellTypeList <- cellTypeList[!is.na(cellTypeList)]
```
Comment (imran-aifi) [inline R/callOpenTiles.R:442]: we don't need this line (442) anymore since cellTypeList is not used, right?

**Review by imran-aifi — CHANGES_REQUESTED** [review]

File: R/callOpenTiles.R (hunk 3 above)
Comment (markphillippebworth) [reply R/callOpenTiles.R:442]: Good point. Yes, it's not needed .

File: R/callOpenTiles.R (hunk 1 above)
Comment (markphillippebworth) [reply R/callOpenTiles.R:386]: It does occur in .callOpenTiles_default, but .callOpenTiles_default isn't run when we use an ArchR project.

Comment (markphillippebworth) [reply R/callOpenTiles.R:386]: Hence the need to check within .callOpenTiles_ArchR

File: R/callOpenTiles.R (hunk 2 above)
Comment (markphillippebworth) [reply R/callOpenTiles.R:407]: I moved it so that it's not on the copy.

**Pull request merged**